components/script/dom/bindings/typedarrays.rs

```diff
@@ -142,3 +161,34 @@ where
         TypedArray::from(dest.get())
     }
 }
+
+pub fn create_new_external_array_buffer<T>(
+    cx: JSContext,
+    mapping: Arc<Mutex<Vec<T::Element>>>,
+    offset: usize,
+    range_size: usize,
+    m_end: usize,
+) -> HeapTypedArray<T>
+where
+    T: TypedArrayElement + TypedArrayElementCreator,
+    T::Element: Clone + Copy,
+{
+    /// `freeFunc()` must be threadsafe, should be safely called from any thread
+    /// without causing conflicts or unexpected behavior.
+    /// <https://github.com/servo/mozjs/blob/main/mozjs-sys/mozjs/js/public/ArrayBuffer.h#L89>
+    unsafe extern "C" fn free_func(_contents: *mut c_void, free_user_data: *mut c_void) {
+        let _ = Arc::from_raw(free_user_data as _);
+    }
+
+    unsafe {
+        let array_buffer = NewExternalArrayBuffer(
+            *cx,
+            range_size as usize,
+            mapping.lock().unwrap().borrow_mut()[offset as usize..m_end as usize].as_mut_ptr() as _,
+            Some(free_func),
+            Arc::into_raw(mapping.clone()) as _,
```

Comment:
There is no need to clone the mapping here(you own the arc passed as argument above, so it must have been cloned before).

Reply:
I think I am forced to clone either for:

'contents : mapping.clone().lock().unwrap().borrow_mut()[offset as usize..m_end as usize].as_mut_ptr() as _,`

of for 

`freeUserData : Arc::into_raw(mapping.clone()) as _, `

else I will get :
`
cannot move out of `mapping` because it is borrowed
move out of `mapping` occurs here
`